components/camel-google/camel-google-bigquery/src/main/java/org/apache/camel/component/google/bigquery/GoogleBigQueryEndpoint.java

```diff
@@ -55,7 +55,14 @@ protected GoogleBigQueryEndpoint(String endpointUri, GoogleBigQueryComponent com
 
     @Override
     public Producer createProducer() throws Exception {
-        BigQuery bigquery = getConfiguration().getConnectionFactory().getDefaultClient();
+        GoogleBigQueryConnectionFactory connFactory = configuration.getConnectionFactory();
+        if (connFactory == null) {
+            connFactory = new GoogleBigQueryConnectionFactory()
```

Comment:
It is better to move such logic (connFactory == null) block to doStart so the producer is not changing endpoint configuration.